src/mesh/unstructured_mesh.C

```diff
@@ -99,6 +99,9 @@ void UnstructuredMesh::copy_nodes_and_elements(const UnstructuredMesh & other_me
     //Preallocate Memory if necessary
     this->reserve_nodes(other_mesh.n_nodes());
 
+#ifdef LIBMESH_ENABLE_UNIQUE_ID
+    unique_id_type potentially_next_unique_id = _next_unique_id;
```

Comment:
This seems like an important and distinct change, doesn't it?  Probably worth backporting?  Is it worth merging separately if this PR gets delayed any further?

Reply:
Yeah, this is not closely related to this PR. Let me have it in a separate one.

Reply:
This was actually a @lindsayad change, [#3013](https://github.com/libMesh/libmesh/pull/3013) (sorry for including it here, I figured it would resolve on rebase anyway)